src/app/user-form/user-form.component.ts

```diff
@@ -0,0 +1,136 @@
+import { Component, EventEmitter, Input, OnChanges, OnInit, Output, SimpleChanges } from '@angular/core';
+import { AbstractControl, FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { User } from '../user';
+import { UserService } from '../user.service';
+import { EmailValidator } from '../email.validator';
+
+@Component({
+	selector: 'app-user-form',
+	templateUrl: './user-form.component.html',
+	styleUrls: ['./user-form.component.css']
+})
+export class UserFormComponent implements OnInit, OnChanges {
+
+	@Input()
+	public formType: "Add" | "Edit" = "Add";
+
+	@Input()
+	public userToEdit: User | undefined = undefined;
+
+	@Output()
+	public submitFormEvent: EventEmitter<{ user: User, formType: "Add" | "Edit" }> = new EventEmitter<{ user: User, formType: "Add" | "Edit" }>();
```

Comment:
```suggestion
	public onSubmit = new EventEmitter<{ user: User, formType: "Add" | "Edit" }>();
```

according to [angular documentation](https://angular.io/guide/styleguide#dont-prefix-output-properties) prefixing with "on-" for eventEmitters is a bad practice. But we declared an own convention:
- prefix with "on" eventEmitters (outputs). For example: onClick, onUserChange, onHighlight
- handlers on these outputs should replace the "on" prefix with "handle". For example: handleClick(), handleuserChange()
- also, we got rid of return type on EventEmitters, because it would be very long and redundant